Cabal-syntax/src/Distribution/Types/Flag.hs

```diff
@@ -290,16 +291,17 @@ parsecFlagAssignment = mkFlagAssignment <$>
 --
 -- @since 3.4.0.0
 parsecFlagAssignmentNonEmpty :: CabalParsing m => m FlagAssignment
-parsecFlagAssignmentNonEmpty = mkFlagAssignment . toList <$>
-    P.sepByNonEmpty (onFlag <|> offFlag) P.skipSpaces1
+parsecFlagAssignmentNonEmpty = mkFlagAssignment <$> some (onFlag <|> offFlag)
   where
     onFlag = do
         _ <- P.char '+'
         f <- parsec
+        _ <- P.skipSpaces
```

Comment:
this allows to parse `+foo+bar`, without a space, doesn't it?

Reply:
I guess it does parse that as the foo flag followed by the bar flag. I personally think that's fine, if a bit quirky. If you have a different suggestion as to how to fix the original bug, which prevents valid parses, I'm open to other ideas.

Reply:
```diff
diff --git a/Cabal-syntax/src/Distribution/Types/Flag.hs b/Cabal-syntax/src/Distribution/Types/Flag.hs
index 713430ee6..8346925aa 100644
--- a/Cabal-syntax/src/Distribution/Types/Flag.hs
+++ b/Cabal-syntax/src/Distribution/Types/Flag.hs
@@ -261,6 +261,14 @@ instance Pretty FlagAssignment where
 -- >>> simpleParsec "+foo -bar baz" :: Maybe FlagAssignment
 -- Nothing
 --
+-- Issue #7279 was fixed in Cabal-3.8
+--
+-- >>> explicitEitherParsec (parsecCommaList parsec) "+foo , -bar" :: Either String [FlagAssignment]
+-- Right [fromList [(FlagName "foo",(1,True))],fromList [(FlagName "bar",(1,False))]]
+--
+-- >>> simpleParsec "+foo+foo" :: Maybe FlagAssignment
+-- Nothing
+--
 -- @since 3.4.0.0
 --
 instance Parsec FlagAssignment where
@@ -272,19 +280,23 @@ dispFlagAssignment = Disp.hsep . map (Disp.text . showFlagValue) . unFlagAssignm
 
 -- | Parses a flag assignment.
 parsecFlagAssignment :: CabalParsing m => m FlagAssignment
-parsecFlagAssignment = mkFlagAssignment <$> many (onFlag <|> offFlag)
+parsecFlagAssignment = mkFlagAssignment <$> sepByEnding (onFlag <|> offFlag) P.skipSpaces1
   where
     onFlag = do
         _ <- P.char '+'
         f <- parsec
-        _ <- P.skipSpaces
         return (f, True)
     offFlag = do
         _ <- P.char '-'
         f <- parsec
-        _ <- P.skipSpaces
         return (f, False)
 
+    sepByEnding :: CabalParsing m => m a -> m b -> m [a]
+    sepByEnding p sep = afterSeparator where
+        element        = (:) <$> p <*> afterElement
+        afterElement   = sep *> afterSeparator <|> pure []
+        afterSeparator = element <|> pure []
+
 -- | Parse a non-empty flag assignment
 --
 -- The flags have to explicitly start with minus or plus.
```

Reply:
Nice 👍